app/src/main/java/com/dcrandroid/fragments/more/PoliteiaFragment.kt

```diff
@@ -0,0 +1,864 @@
+            empty_list.visibility = View.GONE
+        }
+
+        if (loading.get()) {
+            return@launch
+        }
+
+        loading.set(true)
+        val limit = 10
+        val offset = when {
+            loadMore -> proposals.size
+            else -> 0
+        }
+
+        val jsonResult = multiWallet.politeia.getProposals(Constants.PROPOSALS_REJECTED, offset, limit, newestProposalsFirst)
+
+        // Check if the result object from the json response is null
+        val resultObject = JSONObject(jsonResult).get("result")
+        val resultObjectString = resultObject.toString()
+        val rejectedProposalList = if (resultObjectString == "null") {
+            gson.fromJson("[]", Array<Proposal>::class.java)
+        } else {
+            val resultArray = JSONObject(jsonResult).getJSONArray("result")
+            val resultArrayString = resultArray.toString()
+            gson.fromJson(resultArrayString, Array<Proposal>::class.java)
+        }
+
+        initialLoadingDone.set(true)
+
+        if (rejectedProposalList == null) {
+            loadedAll = true
+            loading.set(false)
+            hideLoadingView()
+
+            if (!loadMore) {
+                proposals.clear()
+            }
+            return@launch
+        }
+
+        if (rejectedProposalList.size < limit) {
+            loadedAll = true
+        }
+
+        if (loadMore) {
+            val positionStart = proposals.size
+            proposals.addAll(rejectedProposalList)
+            withContext(Dispatchers.Main) {
+                proposalAdapter?.notifyItemRangeInserted(positionStart, rejectedProposalList.size)
+
+                // notify previous last item to remove bottom margin
+                proposalAdapter?.notifyItemChanged(positionStart - 1)
+            }
+
+        } else {
+            proposals.let {
+                it.clear()
+                it.addAll(rejectedProposalList)
+            }
+
+            checkEmptyProposalList("rejected")
+
+            withContext(Dispatchers.Main) {
+                proposalAdapter?.notifyDataSetChanged()
+            }
+        }
+
+        loading.set(false)
+        hideLoadingView()
+
+        swipe_refresh_layout.isRefreshing = false
+    }
+
+    private fun loadAbandonedProposals(loadMore: Boolean = false) = GlobalScope.launch(Dispatchers.Default) {
+        activity!!.runOnUiThread {
+            showLoadingView()
+            swipe_refresh_layout.isRefreshing = true
+            swipe_refresh_layout.visibility = View.GONE
+            empty_list.visibility = View.GONE
+        }
+
+        if (loading.get()) {
+            return@launch
+        }
+
+        loading.set(true)
+        val limit = 10
+        val offset = when {
+            loadMore -> proposals.size
+            else -> 0
+        }
+
+        val jsonResult = multiWallet.politeia.getProposals(Constants.PROPOSALS_ABANDONED, offset, limit, newestProposalsFirst)
```

Comment:
All the `load-` functions on this page are redundant and should be merged into one. This particular line is the only thing different in each function and you can replace it with a parameter.